server.js

```diff
@@ -16,8 +14,95 @@ const welcomeMessage = {
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
+app.get("/messages/search", (req, res) => {
+  const { term } = req.query;
+
+  console.log(term);
+
+  const filterMessages = messages.filter((message) =>
+    message.text.toLowerCase().includes(term.toLowerCase())
+  );
+
+  console.log(filterMessages);
+
+  res.send(filterMessages);
+});
+
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.get("/messages/latest", (req, res) => {
+  res.send(messages.slice(-10));
+});
+
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+
+  const messageObject = {
+    id: messages.length + 1,
+
+    from,
+
+    text,
+
+    timeSent: new Date().toLocaleDateString(),
+  };
+
+  if (from.length === 0 || text.length === 0) {
+    return res.status(400).send("Please, complete body...");
+  } else {
+    res.json(messageObject);
+  }
+});
+
+app.get("/messages/:id", (req, res) => {
+  const id = req.params.id;
+
+  messages = messages.filter((message) => message.id === Number(id));
+
+  res.status(200).json(messages);
+});
+
+app.get("/messages/search", (req, res) => {
+  let search = req.query.term;
+
+  let searchResult = messages.filter((msg) =>
+    msg.text.toLowerCase().includes(search.toLowerCase())
+  );
+
+  res.send(searchResult);
+});
+
+app.delete("/messages/:id", (req, res) => {
+  const id = req.params.id;
```

Comment:
Just because you've used the `const { x } = req.params` syntax before, try and be consistent with it's use in destructuring